src/towncrier/test/test_check.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Amber Brown, 2017
+# See LICENSE for details.
+
+import os
+
+from twisted.trial.unittest import TestCase
+from click.testing import CliRunner
+from subprocess import call
+
+from towncrier.check import _main
+
+
+class TestChecker(TestCase):
+    maxDiff = None
+
+    def test_no_changes_made(self):
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            with open('pyproject.toml', 'w') as f:
+                f.write(
+                    '[tool.towncrier]\n'
+                    'package = "foo"\n'
+                )
+            os.mkdir('foo')
+            with open('foo/__init__.py', 'w') as f:
+                f.write('__version__ = "1.2.3"\n')
+            os.mkdir('foo/newsfragments')
+            fragment_path = 'foo/newsfragments/123.feature'
+            with open(fragment_path, 'w') as f:
+                f.write('Adds levitation')
+
+            call(["git", "init"])
+            call(["git", "config", "user.name", "user"])
+            call(["git", "config", "user.email", "user@example.com"])
+            call(["git", "add", "."])
+            call(["git", "commit", "-m", "Initial Commit"])
+            call(["git", "checkout", "-b", "otherbranch"])
```

Comment:
These 3 tests have about 20 lines of identical setup code. Maybe factor out a helper function?